lib/board.rb

```diff
@@ -0,0 +1,40 @@
+class Board
+  def initialize
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9] 
+  end
+
+  # def play
+  #   puts draw_board
+  # end
+
+  def draw_board
+    "\n" +
+    " #{@board[0]} | #{@board[1]} | #{@board[2]} " + "\n" +
+    "===+===+===" + "\n" +
+    " #{@board[3]} | #{@board[4]} | #{@board[5]} " + "\n" +
+    "===+===+===" + "\n" +
+    " #{@board[6]} | #{@board[7]} | #{@board[8]} " + "\n" +
+    "\n"
+  end
+
+  def free_spot?(spot)
+    @board[spot - 1] != 'X' && @board[spot - 1] != 'O'
+  end
+
+  def valid_marker?(player_mark)
+    player_mark == 'X' || player_mark == 'O'
+  end
+
+  def mark_board(player_mark, spot)
+    if !valid_marker?(player_mark)      
+      puts "\n I don't recognize #{player_mark}. Try again with X or O.\n"
+    elsif free_spot?(spot)
+      @board[spot - 1] = player_mark
+    else 
+      puts "\n Sorry, that spot is taken. Please try a again.\n"
+    end
+  end
```

Comment:
I see tests for `draw_board` but not for these methods.  Can we flush these tests out? Where these methods even TDD'd (part of the [technical requirements][1] for this project).

[1]: https://github.com/8thlight/apprenticeship_syllabus/blob/master/shared_resources/projects/tic_tac_toe/01_beginner/human_player.md#technical-requirements

Reply:
This is helpful! I saw them as "helper functions" and so I was testing their functionality through the `mark_board` tests I wrote. But I had not written tests for the methods themselves to ensure they were working _before_ adding them into `mark_board`. I'm updating the tests now to ensure these are explicitly tested.

Reply:
Cool. If you ever have functions that are only ever used internal to the class, in Ruby you can use the `private` keyword. These function will not be exposed publicly, and also in general private functions are not unit tested.

There is a caveat; the use of private functions may point to a class hiding within another class, but not always.